pkg/bundle/internal/target/target.go

```diff
@@ -111,10 +111,18 @@ func (r *Reconciler) SyncConfigMap(
 	}
 	configMapBinData := resolvedBundle.BinaryData
 
+	// Create empty map.
+	// If no additional formats are present then
+	// no annotations will be written into ConfigMap
+	var passwdHashAnnotations = make(map[string]string)
+	if bundle.Spec.Target.AdditionalFormats != nil {
+		passwdHashAnnotations = truststorePasswordAnnotations(bundleTarget.AdditionalFormats)
+	}
+
```

Comment:
I suggest you consolidate these optional new annotations with the data hash value (and annotation) a few lines above. This will make the code more compact and also easier to reason about, at least for me.

We should probably also move the calculation of the hashes to the Bundle reconcile function, and supply the desired annotations as a parameter to the sync function. This will avoid calculating the same value for each and every target resource. But that can be fixed in a follow-up PR.

Reply:
Yes, we can improve it in a follow-up PRs.